src/DockComponentsFactory.cpp

```diff
@@ -0,0 +1,68 @@
+//============================================================================
+/// \file   DockComponentsFactory.cpp
+/// \author Uwe Kindler
+/// \date   10.02.2020
+/// \brief  Implementation of DockComponentsFactory
+//============================================================================
+
+//============================================================================
+//                                   INCLUDES
+//============================================================================
+#include <DockComponentsFactory.h>
+#include <QScopedPointer>
+
+#include "DockWidgetTab.h"
+#include "DockAreaTabBar.h"
+#include "DockAreaTitleBar.h"
+#include "DockWidget.h"
+#include "DockAreaWidget.h"
+
+namespace ads
+{
+static QScopedPointer<CDockComponentsFactory> DefaultFactory(new CDockComponentsFactory());
```

Comment:
Just a thought, QScopedPointer is not used anymore since std::unique_ptr is available.

Reply:
+1, BUT: The only reason I can see to stay with `QScopedPointer` is the global policy to stay withing Qt world and involve other libraries as little as possible. I know today's `unique_ptr` is templated (i.e. inlined), but if in some implementation it will use something from run-time libraries then using it will require each user to synch with C++ run-time, which is good pain especially with MS compilers.